meshio/vtk_io.py

```diff
@@ -359,25 +359,59 @@ def translate_cells(data, types, cell_data_raw):
     # See <https://stackoverflow.com/q/47310359/353337> for better
     # alternatives.
     bins = {u: numpy.where(types == u)[0] for u in numpy.unique(types)}
+    has_polygon = meshio_to_vtk_type["polygon"] in bins
 
     # Deduct offsets from the cell types. This is much faster than manually
     # going through the data array. Slight disadvantage: This doesn't work for
     # cells with a custom number of points.
     numnodes = numpy.empty(len(types), dtype=int)
-    for tpe, idx in bins.items():
-        numnodes[idx] = vtk_type_to_numnodes[tpe]
-    offsets = numpy.cumsum(numnodes + 1) - (numnodes + 1)
+    if not has_polygon:
```

Comment:
Negated `if`s with an `else`, e.g.,
```
if not hungry:
    don't eat
else:
    eat
```
are harder to read than
```
if hungry:
    eat
else:
    don't eat
```
Please adapt all conditionals in the PR accordingly.